api/src/types/validation.ts

```diff
@@ -254,3 +255,23 @@ export const loadedEntity = Joi.object({
   attributes: attributes.required(),
   relationships: relationships.required(),
 });
+
+export const ENTITY_API_ALL = "all";
+export const apiEntityTypes = Joi.string().custom((raw: string, helpers) => {
```

Comment:
@mjlangan it may be possible to get the same behavior here with [`Joi.alternatives()`](https://github.com/sideway/joi/blob/master/API.md#alternatives), where one alternative is the string `all`, and the other is a [`Joi.array()`](https://stackoverflow.com/a/42656623/2096369) where the items are of type string and the values are whitelisted to the entity types. Not at all necessary, though could turn this into a a couple of lines if I understand the logic correctly

Reply:
I had trouble trying to figure out how `Joi.array()` expects the query param to look. I knew I wanted to allow either a single string or a comma-separated list, and it was way faster for me to just write a custom function to do it than try to figure out how to express it with Joi's fluent API.